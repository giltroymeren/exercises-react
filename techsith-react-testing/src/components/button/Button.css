.button-main {
	border:  1px solid #CDCDCD;
	padding: 0.5em;
}