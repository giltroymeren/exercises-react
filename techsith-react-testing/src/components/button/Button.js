import React from 'react'

import './Button.css'

function Button({label}) {
	return (
		<div data-testid="button"
			className="button-main">{label}</div>
	)
}

export default Button