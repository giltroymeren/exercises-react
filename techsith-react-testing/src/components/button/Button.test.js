import React from 'react'
import ReactDOM from 'react-dom'
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'

import Button from './Button'

describe('Button', () => {
	const LABEL = 'Click Me!'
	afterEach(cleanup)

	it('renders without crashing', () => {
		const div = document.createElement('div')
		ReactDOM.render(<Button />, div);
	});

	it('renders correctly', () => {
		const { getByTestId } = render(<Button label={LABEL}/>)
		expect(getByTestId('button')).toHaveTextContent(LABEL)
	});

	it('matches snapshot', () => {
		const tree = renderer.create(<Button label={LABEL} />).toJSON()
		expect(tree).toMatchSnapshot()
	})
})