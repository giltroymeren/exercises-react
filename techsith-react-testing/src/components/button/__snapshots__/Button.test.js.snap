// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<div
  className="button-main"
  data-testid="button"
>
  Click Me!
</div>
`;
